src/ds/serializer.h

```diff
@@ -311,8 +311,19 @@ namespace serializer
     template <typename T, typename... Ts>
     static auto deserialize_impl(const uint8_t* data, size_t size)
     {
-      const auto next = std::make_tuple(
-        deserialize_value(data, size, Tag<details::remove_cvref_t<T>>{}));
+      using StrippedT = details::remove_cvref_t<T>;
+
+      if constexpr (
+        std::is_same_v<StrippedT, std::vector<uint8_t>> ||
+        std::is_same_v<StrippedT, ByteRange>)
+      {
+        static_assert(
+          sizeof...(Ts) == 0,
+          "Byte vectors must be the final element in message");
+      }
```

Comment:
This is a vaguely related static assert to enforce a restriction that was a) unenforced, and b) I'd completely forgotten about. We don't size prefix byte ranges, we assume they take up all the space left in the message. We could waste some space by size-prefixing them all, or do a complicated check where only mid-sequence ones are size-prefixed, but since we don't actually have any messages that need this yet here's the easy solution - disallow it with a static_assert.